File: LeetCode/53.Maximum_Subarray/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
t1 = [-2,1,-3,4,-1,2,1,-5,4]
t2 = [1]
t3 = [0]
t4 = [-1]
t5 = [-2147483647]
s = Solution()
if (s.maxSubArray(t1)==6):
    print("Pass")
else:
    print("Test 1 Failed")
if (s.maxSubArray(t2)==1):
    print("Pass")
else:
    print("Test 2 Failed")
if (s.maxSubArray(t3)==0):
    print("Pass")
else:
    print("Test 3 Failed")
if (s.maxSubArray(t4)==-1):
    print("Pass")
else:
    print("Test 4 Failed")
if (s.maxSubArray(t5)==-2147483647):
    print("Pass")
else:
    print("Test 5 Failed")

File: LeetCode/53.Maximum_Subarray/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> nums1 {-2,1,-3,4,-1,2,1,-5,4};
    vector<int> nums2 {1};
    vector<int> nums3 {0};
    vector<int> nums4 {-1};
    vector<int> nums5 {-2147483647};
    if (s.maxSubArray(nums1) == 6) { cout << "Pass" << endl;}
    else { cout << "Failed 1 - t1" << endl;}
    if (s.maxSubArray(nums2) == 1) { cout << "Pass" << endl;}
    else { cout << "Failed 2 - t2" << endl;}
    if (s.maxSubArray(nums3) == 0) { cout << "Pass" << endl;}
    else { cout << "Failed 3 - t3" << endl;}
    if (s.maxSubArray(nums4) == -1) { cout << "Pass" << endl;}
    else { cout << "Failed 4 - t4" << endl;}
    if (s.maxSubArray(nums5) == -2147483647) { cout << "Pass" << endl;}
    else { cout << "Failed 5 - t5" << endl;}
    return 0;
}